django_loci/base/models.py

```diff
@@ -66,12 +70,16 @@ def _validate_geometry_if_not_mobile(self):
     def short_type(self):
         return _(self.type.capitalize())
 
+    # save method is automatically wrapped in atomic transaction
     def save(self, *args, **kwargs):
         # if location type is changed to outdoor, remove all associated floorplans
-        if self.type == 'outdoor' and not self._state.adding:
-            with transaction.atomic():
-                self.objectlocation_set.all().update(floorplan=None, indoor=None)
-                self.floorplan_set.all().delete()
+        if (
+            self.type != self._initial_type
```

Comment:
Let's add also a condition which ensures this is not executed on new objects, should be `not self._state.adding` or something similar.

Reply:
I thought self.type != self._initial_type should do trick but for new objects also this will be true as _initial_type will be None. Adding the `self._state.adding` check back. Thanks